plugins/c8y_log_plugin/src/config.rs

```diff
@@ -0,0 +1,730 @@
+use c8y_smartrest::topic::C8yTopic;
+use mqtt_channel::Message;
+use serde::Deserialize;
+use std::borrow::Borrow;
+use std::collections::HashSet;
+use std::fs;
+use std::hash::{Hash, Hasher};
+use std::path::{Path, PathBuf};
+use tracing::{info, warn};
+
+const DEFAULT_PLUGIN_CONFIG_TYPE: &str = "c8y-log-plugin";
+
+#[derive(Deserialize, Debug, Default)]
+#[serde(deny_unknown_fields)]
+struct RawPluginConfig {
+    pub files: Vec<RawFileEntry>,
+}
+
+#[derive(Deserialize, Debug, Default, PartialEq)]
+#[serde(deny_unknown_fields)]
+pub struct RawFileEntry {
+    pub path: String,
+    #[serde(rename = "type")]
+    config_type: Option<String>,
+}
+
+#[derive(Debug, Eq, PartialEq, Default, Clone)]
+pub struct PluginConfig {
```

Comment:
Do you really need both `RawPluginConfig` and `PluginConfig`. I understand that one holds a `HashSet` and the other holds a `Vec`. I feel just keeping the one with the `Vec` is sufficient. The `HashSet` based one doesn't really add much value as we're going to allow duplicate `type` entries anyway.

Reply:
We didn't agree as a team on that, so I didn't continue down this route.